services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/tasks/RsuMonitoringTask.java

```diff
@@ -0,0 +1,106 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import us.dot.its.jpo.geojsonconverter.partitioner.RsuIntersectionKey;
+import us.dot.its.jpo.ode.api.ConflictMonitorApiProperties;
+import us.dot.its.jpo.ode.api.models.postgres.derived.RsuCredentials;
+import us.dot.its.jpo.ode.api.models.snmp.OIDMap;
+import us.dot.its.jpo.ode.api.models.snmp.RsuState;
+import us.dot.its.jpo.ode.api.services.KafkaProducerService;
+import us.dot.its.jpo.ode.api.services.PostgresService;
+import us.dot.its.jpo.ode.api.services.SNMPService;
+
+@Component
+@ConditionalOnProperty(name = "enable.monitoring", havingValue = "true", matchIfMissing = false)
+public class RsuMonitoringTask {
+
+    private static final Logger log = LoggerFactory.getLogger(RsuMonitoringTask.class);
+
+    private SNMPService snmpService;
+    private KafkaProducerService kafkaService;
+    private PostgresService postgresService;
+    private ConflictMonitorApiProperties properties;
+
+    @Autowired
+    public RsuMonitoringTask(
+            SNMPService snmpService,
+            KafkaProducerService kafkaService,
+            PostgresService postgresService,
+            ConflictMonitorApiProperties properties) {
+        this.snmpService = snmpService;
+        this.kafkaService = kafkaService;
+        this.postgresService = postgresService;
+        this.properties = properties;
+    }
+
+    @Scheduled(fixedRate = 300000) // Check RSU units every 5 minutes
+    public void queryRSUStats() {
+
+        List<RsuCredentials> credentials = postgresService.getRsusWithCredentials();
+
+        for (RsuCredentials cred : credentials) {
+
+            String username = cred.getUsername();
+            String password = cred.getPassword();
+            String encPass = cred.getEncrypt_password();
+            String ip = cred.getIpv4_address();
+            String intersectionId = cred.getIntersection_id();
+
+            log.info("Pulling SNMP Status for RSU: " + ip + " IntersectionID: " + intersectionId);
+
+            if (username == null || password == null || ip == null) {
+                log.warn("Cannot pull data from RSU unit. Missing Username, Password, or IP address. RSU ID: " + ip
+                        + " Intersection ID: " + intersectionId);
+                continue;
+            }
+
+            // enc password is not defined for all RSU units. Try using normal password
+            if (encPass == null) {
+                encPass = password;
+            }
+
+            RsuState state = new RsuState();
+            state.timestamp = Instant.now().toEpochMilli();
+            state.rsuIP = ip;
+            state.intersectionID = intersectionId;
+
+            state.uptime = getIntOID(ip, username, password, encPass, OIDMap.oids.get("rsuModeStatus").getOid());
```

Comment:
(question) How will this handle other vendor models where the OID values for specific metrics may be different? @drewjj  is this something we should consider?

Reply:
This doesn't really treat any individual OID as anything besides a data point. We would need to include the RSU 4.1 OIDs for those that require them. It may be necessary to include more logic around specific OIDs when performing snmpset commands since corner cases are very plentiful when it comes to RSU SNMP support.